src/sagemaker_containers/env.py

```diff
@@ -598,6 +598,38 @@ def framework_module(self):  # type: () -> str
                 my_module:main"""
         return self._framework_module
 
+    def write_new_file(self, dir_name, file_name, file_content=None):  # type: (str, str, str) -> None
+        """Create a new file with customized content.
+        Args:
+            dir_name: name of the directory
+            file_name: name of the file
+            file_content: content of file
+        """
+        file_path = os.path.join(dir_name, file_name)
+        if os.path.exists(file_path):
+            warning_msg = 'File {} already exists and will be overwritten! ' \
+                          'Please check if file with the same name is created before this call.' \
+                .format(file_path)
+            logger.warning(warning_msg)
+
+        with open(file_path, 'w') as f:
```

Comment:
I was actually not assuming to limit it to only written in creation.

It's all about how likely this write file function will be reused in other scenarios besides sucess/failure file. If not, then we can just limit it to mode=x.

Reply:
Sure. So I removed all unnecessary logic in file I/O. 

But I didn't use mode x. First we don't need limit it here since it's very unlikely that files with same name created before. Second mode x is only available in python 3.